In [13]:
#load dependencies
import pandas as pd
import os 

In [14]:
#load dataset
data = pd.read_csv("USA_Congress_Transactions_data.csv")
print(data.head(3))

              amount                                  asset_description  \
0   $1,001 - $15,000                                             BP plc   
1   $1,001 - $15,000                            Exxon Mobil Corporation   
2  $15,001 - $50,000  Industrial Logistics Properties Trust - Common...   

   cap_gains_over_200_usd disclosure_date  disclosure_year district  owner  \
0                   False      10/04/2021             2021     NC05  joint   
1                   False      10/04/2021             2021     NC05  joint   
2                   False      10/04/2021             2021     NC05  joint   

                                            ptr_link      representative  \
0  https://disclosures-clerk.house.gov/public_dis...  Hon. Virginia Foxx   
1  https://disclosures-clerk.house.gov/public_dis...  Hon. Virginia Foxx   
2  https://disclosures-clerk.house.gov/public_dis...  Hon. Virginia Foxx   

  ticker transaction_date      type  
0     BP       2021-09-27  purchase  
1    

In [15]:
print(data.columns)
print(data.shape)

Index(['amount', 'asset_description', 'cap_gains_over_200_usd',
       'disclosure_date', 'disclosure_year', 'district', 'owner', 'ptr_link',
       'representative', 'ticker', 'transaction_date', 'type'],
      dtype='object')
(15433, 12)


**Fix Columns and Stadarize Values**

In [16]:
#Analyze uniqueness or rows 
print(data.amount.unique())

['$1,001 - $15,000' '$15,001 - $50,000' '$50,001 - $100,000'
 '$100,001 - $250,000' '$1,001 -' '$250,001 - $500,000'
 '$500,001 - $1,000,000' '$5,000,001 - $25,000,000'
 '$1,000,001 - $5,000,000' '$1,000,000 +' '$1,000 - $15,000'
 '$15,000 - $50,000' '$50,000,000 +' '$1,000,000 - $5,000,000']


There is values in the amount column starting with a 1 dollar difference
'$1,000 - $15,000' and '$1,001 - $15,000' should be the same value 
lets fix for the difference across the entire column 

In [17]:
#Fix amount column
for i in range(0,len(data['amount'])):
    index = data['amount'][i].find('01')
    if index != -1:
      data['amount'][i] = list(data['amount'][i])
      data['amount'][i][index:index+2] = '00'
      data['amount'][i] = ''.join(data['amount'][i])

C:\Users\jhoel\AppData\Local\Temp\ipykernel_14768\2794293020.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['amount'][i] = list(data['amount'][i])
C:\Users\jhoel\AppData\Local\Temp\ipykernel_14768\2794293020.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['amount'][i] = ''.join(data['amount'][i])


In [ ]:
print(data.amount.unique())
print(data.shape)

['$1,000 - $15,000' '$15,000 - $50,000' '$50,000 - $100,000'
 '$100,000 - $250,000' '$1,000 -' '$250,000 - $500,000'
 '$500,000 - $1,000,000' '$5,000,000 - $25,000,000'
 '$1,000,000 - $5,000,000' '$1,000,000 +' '$50,000,000 +']
(15433, 12)


In [ ]:
#Delete rows with incomplete data 
data = data.dropna()
print(data.shape)

(9521, 12)


In [ ]:
#Analysis distric, representative, and type 
data.district.unique()

array(['NC05', 'CA47', 'VA04', 'NV03', 'AL05', 'TX10', 'WA01', 'MT00',
       'OR03', 'KY01', 'FL21', 'RI02', 'UT03', 'AZ03', 'NJ05', 'MA05',
       'WV01', 'CA12', 'TN09', 'VA11', 'NC04', 'NC06', 'CA52', 'FL17',
       'VA08', 'PA05', 'TN03', 'PA16', 'IL03', 'TN07', 'VA01', 'MS03',
       'TX17', 'CA17', 'TX35', 'NY06', 'GA12', 'MI12', 'OK04', 'FL02',
       'CO05', 'IN01', 'FL16', 'GA14', 'CA06', 'MN03', 'VT00', 'NY08',
       'CO07', 'CA19', 'TX11', 'FL18', 'IA03', 'OK01', 'IN03', 'MO02',
       'MD08', 'FL15', 'IL10', 'WA08', 'MI06', 'OH16', 'CA39', 'CA53',
       'KY03', 'TX03', 'FL23', 'CT02', 'PA03', 'SC07', 'OH07', 'TX26',
       'KS01', 'ME01', 'NC02', 'FL12', 'NJ11', 'KS04', 'WI08', 'NJ06',
       'NY27', 'MA06', 'FL14', 'IA01', 'ID02', 'TN01', 'OH05', 'MA03',
       'AZ01', 'GA01', 'NC07', 'IL16', 'FL27', 'SC04', 'IL17', 'TX24',
       'WV03', 'NY12', 'GA08', 'PA09', 'MD06', 'FL04', 'AR02', 'MA04',
       'CA03', 'IN05', 'TN08', 'VA07', 'CA38', 'NY25', 'CA28', 'CA48',
      

In [ ]:
data.representative.unique()

array(['Hon. Virginia Foxx', 'Hon. Alan S. Lowenthal',
       'Hon. Aston Donald McEachin', 'Hon. Susie Lee', 'Hon. Mo Brooks',
       'Hon. Michael T. McCaul', 'Hon. Suzan K. DelBene',
       'Hon. Greg Gianforte', 'Hon. Earl Blumenauer', 'Hon. James Comer',
       'Hon. Lois Frankel', 'Hon. James R. Langevin', 'Hon. John Curtis',
       'Hon. Raúl M. Grijalva', 'Hon. Josh Gottheimer',
       'Hon. Katherine M. Clark', 'Hon. David B. McKinley',
       'Hon. Nancy Pelosi', 'Hon. Steve Cohen', 'Hon. Gerald E. Connolly',
       'Hon. David E. Price', 'None Kathy Manning',
       'Hon. Scott H. Peters', 'Hon. Greg Steube',
       'Hon. Donald Sternoff Honorable Beyer', 'Hon. Mary Gay Scanlon',
       'Hon. Charles J. "Chuck" Fleischmann', 'Hon. Mike Kelly',
       'Hon. Marie Newman', 'Hon. Mark Dr Green',
       'Hon. Robert J. Wittman', 'Hon. Michael Patrick Guest',
       'Hon. Pete Sessions', 'Hon. Rohit Khanna', 'Hon. Lloyd Doggett',
       'Hon. Grace Meng', 'Hon. Richard W. Allen',

In [ ]:
data.type.unique()

array(['purchase', 'sale_partial', 'sale_full', 'exchange', 'sale'],
      dtype=object)

In [ ]:
#Check for duplicate rows 
data.drop_duplicates()
print(data.shape)

(9521, 12)


Now having cleaned this dataset I would continue my analysis on Tableu Public.

I will search for a dataset of the political parties of each representative to compare Republican and Democratic Investments. 

In [ ]:
data.to_csv(".\Cleaned_House_Stock_US.csv",index=False)